components/button/button.tsx

```diff
@@ -44,198 +75,257 @@ const defaultProps = {
+          box-shadow: ${shadow ? theme.expressiveness.shadowSmall : 'none'};
+          --zeit-ui-button-padding: ${padding};
+          --zeit-ui-button-height: ${height};
+          --zeit-ui-button-color: ${colors.color};
+          --zeit-ui-button-bg: ${colors.bg};
+        }
+
+        .tag:hover,
+        .btn:hover {
+          color: ${hoverColors.color};
+          --zeit-ui-button-color: ${hoverColors.color};
+          background-color: ${hoverColors.bg};
+          border-color: ${hoverColors.border};
+          cursor: ${cursor};
+          pointer-events: ${events};
+          box-shadow: ${shadow ? theme.expressiveness.shadowMedium : 'none'};
+          transform: translate3d(0px, ${shadow ? '-1px' : '0px'}, 0px);
+        }
+
+        .tag.active,
+        .tag:active,
+        .btn:active {
+          color: ${activeColors.color};
+          --zeit-ui-button-color: ${activeColors.color};
+          background-color: ${activeColors.bg};
+          border-color: ${activeColors.border};
+          cursor: ${cursor};
+          pointer-events: ${events};
+          box-shadow: ${shadow ? theme.expressiveness.shadowMedium : 'none'};
+          transform: translate3d(0px, ${shadow ? '-1px' : '0px'}, 0px);
+        }
+
+        .tag :global(.text),
+        .btn :global(.text) {
+          position: relative;
+          z-index: 1;
+          display: inline-flex;
+          justify-content: center;
+          align-items: center;
+          text-align: center;
+          line-height: inherit;
+          top: -2px;
+        }
+
+        .tag :global(.text p),
+        .tag :global(.text pre),
+        .tag :global(.text div),
+        .btn :global(.text p),
+        .btn :global(.text pre),
+        .btn :global(.text div) {
+          margin: 0;
+        }
+        .tag :global(.hidden),
+        .btn :global(.hidden) {
+          visibility: hidden;
+        }
+      `}</style>
+    </ButtonOrTag>
+  )
+}
+
+const Button = React.forwardRef<HTMLButtonElement, React.PropsWithChildren<ButtonProps>>(
+  TagOrButtonRender,
+)
+const TagBase = React.forwardRef<HTMLSpanElement, React.PropsWithChildren<TagProps>>(
+  TagOrButtonRender,
 )
 
 type ButtonComponent<T, P = {}> = React.ForwardRefExoticComponent<
   PropsWithoutRef<P> & RefAttributes<T>
 >
-type ComponentProps = Partial<typeof defaultProps> &
-  Omit<Props, keyof typeof defaultProps> &
-  NativeAttrs
+type ComponentProps = Partial<typeof defaultButtonProps> &
+  Omit<ButtonBaseProps, keyof typeof defaultButtonProps> &
+  ButtonNativeAttrs
+
+Button.defaultProps = defaultButtonProps
+
+type TagComponent<T, P = {}> = React.ForwardRefExoticComponent<
+  PropsWithoutRef<P> & RefAttributes<T>
+>
 
-Button.defaultProps = defaultProps
+TagBase.defaultProps = defaultTagProps
 
 export default React.memo(Button) as ButtonComponent<HTMLButtonElement, ComponentProps>
+export const Tag = React.memo(TagBase) as TagComponent<HTMLSpanElement, ComponentProps>
```

Comment:
The `button` and `tag` are the same styles, but it's a little weird to put them in one file, however, it doesn't matter.

Reply:
I'll just leave it be till the file grows too big to maintain.